invenio_pidrelations/serializers/schemas.py

```diff
@@ -101,24 +103,19 @@ def dump_is_last(self, obj):
 
     def dump_type(self, obj):
         """Dump the text name of the relation."""
-        return resolve_relation_type_config(obj.relation_type).name
+        if not isinstance(obj.relation_type, RelationType):
+            return resolve_relation_type_config(obj.relation_type).name
+        else:
+            return obj.relation_type.name
 
     def dump_parent(self, obj):
         """Dump the parent of a PID."""
-        return self._dump_relative(obj.parent)
+        if not self._is_parent(obj):
```

Comment:
Is this really doing the same as the previous method? It seems that it would not dump the parent of a child PIDNode.

Reply:
Done